crates/sui-indexer-alt-jsonrpc/src/config.rs

```diff
@@ -279,10 +284,26 @@ impl CoinsLayer {
     }
 }
 
-impl WriteLayer {
-    pub fn finish(self, base: WriteConfig) -> WriteConfig {
-        check_extra("write", self.extra);
-        WriteConfig {
+impl NodeConfig {
+    pub fn client(&self, fullnode_rpc_url: url::Url) -> anyhow::Result<HttpClient> {
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            CLIENT_SDK_TYPE_HEADER,
+            HeaderValue::from_str(&self.header_value)?,
+        );
+
+        HttpClientBuilder::default()
+            .max_request_size(self.max_request_size)
+            .set_headers(headers.clone())
+            .build(&fullnode_rpc_url)
+            .context("Failed to initialize fullnode RPC client")
+    }
+}
+
+impl NodeLayer {
+    pub fn finish(self, base: NodeConfig) -> NodeConfig {
+        check_extra("node", self.extra);
```

Comment:
@amnn changing the field name here from `write` to `node`, might need to change existing toml files if any.

Reply:
Yep that's right, (let's create that PR as well).